pkg/kv/kvserver/loqrecovery/collect.go

```diff
@@ -62,3 +76,68 @@ func CollectReplicaInfo(
 	}
 	return loqrecoverypb.NodeReplicaInfo{Replicas: replicas}, nil
 }
+
+// checkUncommittedDescriptorsInRaftLog iterates over committed but not applied
+// part of raft log and searches for writes to range descriptor. If any value
+// referencing local range descriptor key is found then we have a potential
+// pending update.
+func checkUncommittedDescriptorsInRaftLog(
+	rangeID roachpb.RangeID,
+	rangeKey roachpb.RKey,
```

Comment:
call this `startKey` because that's what it is, it isn't really clear what a "range key" is.